examples/generate.go

```diff
@@ -0,0 +1,234 @@
+			httpsign.Headers("@method", "@request-target", "workload-identity-token"))
+		if err != nil {
+			return fmt.Errorf("failed to create request signer: %w", err)
+		}
+		signatureInput, signature, err := httpsign.SignRequest("wimse", *signer, req)
+		if err != nil {
+			return fmt.Errorf("failed to sign request: %w", err)
+		}
+
+		*outSignatureInput = signatureInput
+		*outSignature = signature
+		return nil
+	}
+}
+
+func withHttpMessageSigResponse(req *http.Request, resp *http.Response, outSignatureInput *string, outSignature *string) outputOption {
+	return func(result *exampleOutput) error {
+		resp.Header["Workload-Identity-Token"] = []string{result.WIT}
+
+		sigConfig := httpsign.NewSignConfig()
+		sigConfig.SetTag("wimse-workload-to-workload")
+		sigConfig.SetNonce("abcd2222")
+		sigConfig.SignAlg(false)
+		sigConfig.SetExpires(time.Now().Add(time.Minute).Unix())
+		sigHeaders := httpsign.Headers("@status", "workload-identity-token", "content-type", "content-digest")
+		sigHeaders.AddHeaderExt("@method", false, false, true, false)
+		sigHeaders.AddHeaderExt("@request-target", false, false, true, false)
+		signer, err := httpsign.NewEd25519Signer(result.PrivateKey, sigConfig, sigHeaders)
+		if err != nil {
+			return fmt.Errorf("failed to create response signer: %w", err)
+		}
+
+		signatureInput, signature, err := httpsign.SignResponse("wimse", *signer, resp, req)
+		if err != nil {
+			return fmt.Errorf("failed to sign response: %w", err)
+		}
+
+		*outSignatureInput = signatureInput
+		*outSignature = signature
+		return nil
+	}
+}
+
+type WPTParameters struct {
+	Rand             io.Reader
+	Audience         string
+	AccessToken      string
+	TransactionToken string
+	OtherTokens      map[string]string
+}
+
+func withWorkloadProofToken(params WPTParameters, outWPT *string, outWPTClaims *wptClaims) outputOption {
+	return func(result *exampleOutput) error {
+		claims := wptClaims{
+			Aud:       params.Audience,
+			ExpiresAt: now.Add(wptTTL).Unix(),
+			Id:        mustGenerateJTI(params.Rand),
+		}
+
+		if params.AccessToken != "" {
+			claims.Ath = base64UrlEncTokenHash(params.AccessToken)
+		}
+
+		if params.TransactionToken != "" {
+			claims.Tth = base64UrlEncTokenHash(params.TransactionToken)
+		}
+
+		oth := make(map[string]string)
+		for k, v := range params.OtherTokens {
+			oth[k] = base64UrlEncTokenHash(v)
+		}
+		claims.Oth = oth
+
+		signer, err := jose.NewSigner(jose.SigningKey{Algorithm: result.Alg, Key: result.PrivateKey}, (&jose.SignerOptions{}).WithType(wptJWTType))
+		if err != nil {
+			return fmt.Errorf("failed to create WPT signer: %w", err)
+		}
+
+		enc, err := jwt.Signed(signer).Claims(claims).Serialize()
+		if err != nil {
+			return fmt.Errorf("failed to sign WPT: %w", err)
+		}
+
+		*outWPT = enc
+		*outWPTClaims = claims
+		return nil
+	}
+}
+
+func mustGenerateJTI(rand io.Reader) string {
+	b := make([]byte, 16)
+	if _, err := rand.Read(b); err != nil {
+		panic(err)
+	}
+	return hex.EncodeToString(b)
```

Comment:
hex is subjectively aesthetically objectionable and objectively inefficient from a size/space usage